tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -107,7 +114,8 @@ def evaluate_trailer(datafile, trailer_count, multiple_trailer_errors, is_last_l
 def rollback_records(unsaved_records, datafile):
     """Delete created records in the event of a failure."""
     logger.info("Rolling back created records.")
-    for model in unsaved_records:
+    for document in unsaved_records:
+        model = document.Django.model
         num_deleted, models = model.objects.filter(datafile=datafile).delete()
```

Comment:
out of curiosity i ran the big `fake_rollback` file and noticed the delete requests going crazy

https://github.com/raft-tech/TANF-app/assets/111305129/302b5565-3979-44c7-b776-0fb0505feaf4

but the db and elastic eventually both end up in a consistent state.

this may be a topic for another ticket, but implementing some sort of `bulk_delete` may be helpful. existing solutions might not honor the signals needed for elastic, though. https://stackoverflow.com/a/36935536

Reply:
Agreed, our transaction-less rollback is a pain point for sure. It would be interesting to test out the `raw_delete`.

Reply:
have to be careful with ```raw_delete``` though!